src/bot/messaging.rs

```diff
@@ -61,6 +71,71 @@ impl Bot {
         self.post_message(&path, &params)
     }
 
+    /// API call whick will send the requested chat action to your bot.
+    pub fn send_chat_action(&self, chat_id: &i64, action: &ChatAction) -> Result<bool> {
+        let action = get_chat_action(action);
+        println!("{:?}", &action);
+        let path = ["sendChatAction"];
+        let url = ::construct_api_url(&self.bot_url, &path);
+        let params = [("chat_id", chat_id.to_string()),
+            ("action", action)];
+        let mut data = self.client.post(&url).form(&params).send()?;
+        println!("{:?}", &data);
+
+        let rjson: Value = check_for_error(data.json()?)?;
+        let message_json = rjson.get("result").ok_or(JsonNotFound)?;
+        println!("{:?}", message_json);
+        if message_json.as_bool().unwrap() {
+            Ok(true)
+        } else {
+            Ok(false)
+        }
+    }
+
+    /// API call which will send the given contact.
+    pub fn send_contact(&self,
+                        chat_id: &i64,
+                        contact: &Contact,
+                        disable_notification: Option<&bool>,
+                        reply_to_message_id: Option<&i32>,
+                        reply_markup: Option<&Markup>) -> Result<bool> {
+
+        let chat_id: &str = &chat_id.to_string();
+        let disable_notification: &str = &disable_notification.unwrap_or(&false).to_string();
+        let reply_to_message_id: &str = &reply_to_message_id.map(|i| i.to_string())
+            .unwrap_or("None".to_string());
+        let reply_markup =
+            if reply_markup.is_some() {
+                serde_json::to_string(&reply_markup.unwrap())
+                    .unwrap_or("None".to_string())
+            } else {
+                "".to_string()
+            };
+
+        let path = ["sendContact"];
+        let url = ::construct_api_url(&self.bot_url, &path);
+        let params = [("chat_id", chat_id),
+            ("phone_number", &contact.phone_number),
+            ("first_name", &contact.first_name),
+            ("last_name", &contact.clone().last_name.unwrap()),
+            ("disable_notification", disable_notification),
+            ("reply_to_message_id", reply_to_message_id),
+            ("reply_markup", &reply_markup)];
+        //println!("{:?}", params);
+        let mut data = self.client.post(&url).form(&params).send()?;
+        println!("{:?}", &data);
+
+        let rjson: Value = check_for_error(data.json()?)?;
+        println!("{:?}", &rjson);
+        let message_json = rjson.get("result").ok_or(JsonNotFound)?;
+        //println!("{:?}", message_json);
+        if message_json.as_bool().unwrap_or(false) {
```

Comment:
Please look at my other methods and refer to [this part](https://core.telegram.org/bots/api#sendcontact) of the Telegram bot API docs. It should return the sent message on success, otherwise it should throw an error. Returning `Ok` means something has gone correct. This isn't the case. Like the Rust docs say: "Result is a type that represents either success (Ok) or failure (Err)."